From 0e2b79422fe7da402316cf3018ee8366822f9500 Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Sun, 14 Jul 2024 08:03:03 +0800
Subject: [PATCH 2/2] 7max

---
 CPP/7zip/UI/Console/Main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index 4fa5c35..8a2f7f9 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -848,6 +848,7 @@ int Main2(
   UStringVector commandStrings;
   
   #ifdef _WIN32
+  LoadLibraryEx(L"7max.dll", NULL, LOAD_LIBRARY_SEARCH_APPLICATION_DIR);
   NCommandLineParser::SplitCommandLine(GetCommandLineW(), commandStrings);
   #else
   {
-- 
2.45.2.windows.1

