--- a/Config.vcxproj
+++ b/Config.vcxproj
@@ -38,7 +38,7 @@
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       
       <!-- default -->
-      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
       
     </ClCompile>
   </ItemDefinitionGroup>
--- a/source/script2.cpp
+++ b/source/script2.cpp
@@ -546,6 +546,11 @@

 void LaunchAutoHotkeyUtil(LPTSTR aFile, bool aIsScript)
 {
+	TCHAR prefix[35] = _T("..\\share\\autohotkey\\");
+	size_t n = _tcslen(g_script.mOurEXE);
+	if (n >= 29 && !_tcsicmp(g_script.mOurEXE+(n-27), _T("\\Library\\bin\\AutoHotkey.exe"))) {
+		aFile = _tcscat(prefix, aFile);
+	}
 	LPTSTR file = aFile, args = _T("");
 	bool our_file, result = false;
 	if (!FindAutoHotkeyUtil(aFile, our_file))
