--- a/build/dynamic.rs
+++ b/build/dynamic.rs
@@ -152,6 +152,7 @@
         // The official LLVM build uses `libclang.dll` on Windows instead of
         // `clang.dll`. However, unofficial builds such as MinGW use `clang.dll`.
         files.push("libclang.dll".into());
+        files.push("libclang-*.dll".into());
     } else if target_os!("cygwin") {
         files.push("msys-clang-*.dll".into());
         files.push("cygclang-*.dll".into());
