--- a/base/shell/cmd/CMakeLists.txt
+++ b/base/shell/cmd/CMakeLists.txt
@@ -60,9 +60,10 @@
     precomp.h)

 add_rc_deps(cmd.rc ${CMAKE_CURRENT_SOURCE_DIR}/res/terminal.ico)
-add_executable(cmd ${SOURCE} cmd.rc)
+# BUG: cmd.rc doesn't work in DLL
+add_library(cmd SHARED ${SOURCE} libcmd.def)
 add_pch(cmd precomp.h SOURCE)
-set_module_type(cmd win32cui UNICODE)
+set_module_type(cmd win32dll UNICODE)
 target_link_libraries(cmd conutils wine)
 add_importlibs(cmd advapi32 msvcrt kernel32 ntdll)
 add_delay_importlibs(cmd user32)
--- a/base/shell/cmd/cmd.c
+++ b/base/shell/cmd/cmd.c
@@ -153,8 +153,8 @@
 BOOL bCanExit = TRUE;     /* Indicates if this shell is exitable */
 BOOL bCtrlBreak = FALSE;  /* Ctrl-Break or Ctrl-C hit */
 BOOL bIgnoreEcho = FALSE; /* Set this to TRUE to prevent a newline, when executing a command */
-static BOOL fSingleCommand = 0; /* When we are executing something passed on the command line after /C or /K */
-static BOOL bAlwaysStrip = FALSE;
+DECLSPEC_EXPORT BOOL fSingleCommand = 0; /* When we are executing something passed on the command line after /C or /K */
+DECLSPEC_EXPORT BOOL bAlwaysStrip = FALSE;
 INT  nErrorLevel = 0;     /* Errorlevel of last launched external program */
 CRITICAL_SECTION ChildProcessRunningLock;
 BOOL bDisableBatchEcho = FALSE;
@@ -1768,7 +1768,7 @@
     return SubstituteVars(ip, commandline, _T('%'));
 }

-static INT
+DECLSPEC_EXPORT INT
 ProcessInput(VOID)
 {
     INT Ret = 0;
@@ -2069,7 +2069,7 @@
 }

 /* Get the command that comes after a /C or /K switch */
-static VOID
+DECLSPEC_EXPORT VOID
 GetCmdLineCommand(
     OUT LPTSTR commandline,
     IN LPCTSTR ptr,
@@ -2129,7 +2129,7 @@
  * Set up global initializations and process parameters.
  * Return a pointer to the command line if present.
  */
-static LPCTSTR
+DECLSPEC_EXPORT LPCTSTR
 Initialize(VOID)
 {
     HMODULE NtDllModule;
@@ -2337,7 +2337,7 @@
 }


-static VOID Cleanup(VOID)
+DECLSPEC_EXPORT VOID Cleanup(VOID)
 {
     /* Run cmdexit.bat */
     if (IsExistingFile(_T("cmdexit.bat")))
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,10 @@

 project(REACTOS)

+include(CheckIPOSupported)
+check_ipo_supported()
+
+set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_INCLUDE_DIRECTORIES_PROJECT_BEFORE ON)
 set(CMAKE_SHARED_LIBRARY_PREFIX "")
--- a/sdk/include/asm/CMakeLists.txt
+++ b/sdk/include/asm/CMakeLists.txt
@@ -3,6 +3,7 @@

 set_entrypoint(genincdata 0)
 add_dependencies(genincdata bugcodes psdk xdk)
+set_target_properties(genincdata PROPERTIES INTERPROCEDURAL_OPTIMIZATION OFF)

 if(MSVC)
     set(OPT_MS "-ms")
