From b9a9c3be7b4fa3195b994f184c63f9e6137f940a Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Wed, 25 Oct 2023 19:01:51 +0800
Subject: [PATCH] conda build

---
 src/utils/FileUtil.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/FileUtil.cpp b/src/utils/FileUtil.cpp
index 16d9ed9..b94de61 100644
--- a/src/utils/FileUtil.cpp
+++ b/src/utils/FileUtil.cpp
@@ -473,7 +473,7 @@ char* GetPathOfFileInAppDir(const char* fileName) {
     }
     WCHAR* moduleDir = path::GetDirTemp(modulePath);
     WCHAR* fileNameW = ToWstrTemp(fileName);
-    WCHAR* path = path::JoinTemp(moduleDir, fileNameW);
+    WCHAR* path = path::JoinTemp(moduleDir, L"..\\etc", fileNameW);
     path = path::Normalize(path);
     char* res = ToUtf8(path);
     str::Free(path);
-- 
2.42.0.windows.1

