From 3d2b60532f69e0775312037d133a7800855db239 Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Mon, 25 Sep 2023 18:58:42 +0800
Subject: [PATCH 2/2] msvc 17.3 compat

---
 src/CrashHandler.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/CrashHandler.cpp b/src/CrashHandler.cpp
index f2b92aa..e4b484b 100644
--- a/src/CrashHandler.cpp
+++ b/src/CrashHandler.cpp
@@ -732,7 +732,8 @@ void InstallCrashHandler(const WCHAR* crashDumpPath, const WCHAR* crashFilePath,
     signal(SIGABRT, onSignalAbort);
 #if COMPILER_MSVC
     ::set_terminate(onTerminate);
-    ::set_unexpected(onUnexpected);
+    // https://github.com/sumatrapdfreader/sumatrapdf/commit/c507fd6eb52c37f513167f4e3594b920d0a6b779
+    // ::set_unexpected(onUnexpected);
 #endif
 }
 
-- 
2.42.0.windows.1

