From ecf3d275add404c4096c16c74f062af18249562c Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Mon, 25 Sep 2023 19:10:16 +0800
Subject: [PATCH] conda build

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43ac22e..12c2fbd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,8 +188,8 @@ if(ENABLE_LTO)
   if(USING_GCC)
     jerry_add_compile_flags(-fno-fat-lto-objects)
     # Use gcc-ar and gcc-ranlib to support LTO
-    set(CMAKE_AR "gcc-ar")
-    set(CMAKE_RANLIB "gcc-ranlib")
+    # set(CMAKE_AR "gcc-ar")
+    # set(CMAKE_RANLIB "gcc-ranlib")
   endif()
   if(USING_TI)
     jerry_add_link_flags(-lto)
-- 
2.42.0.windows.1

