From 37c4df826e0ed9f3c31efbbbb0736f74d57105e8 Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Sat, 8 Jun 2024 09:06:58 +0800
Subject: [PATCH 2/2] enable lto

---
 CMakeLists.txt | 22 +++-------------------
 1 file changed, 3 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 314d5ad..3bb09b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,12 +119,6 @@ if(USING_MSVC)
   set(ENABLE_STRIP_MESSAGE       " (FORCED BY COMPILER)")
 endif()
 
-if(CYGWIN OR MINGW OR MSYS)
-  set(ENABLE_LTO         OFF)
-
-  set(ENABLE_LTO_MESSAGE         " (FORCED BY PLATFORM)")
-endif()
-
 # Generate compile_commands.json
 set(CMAKE_EXPORT_COMPILE_COMMANDS ${ENABLE_COMPILE_COMMANDS})
 
@@ -190,19 +184,9 @@ jerry_add_flags(CMAKE_EXE_LINKER_FLAGS ${FLAGS_COMMON_ARCH})
 
 # LTO
 if(ENABLE_LTO)
-  if(USING_GCC OR USING_CLANG)
-    jerry_add_compile_flags(-flto)
-    jerry_add_link_flags(-flto)
-  endif()
-  if(USING_GCC)
-    jerry_add_compile_flags(-fno-fat-lto-objects)
-    # Use gcc-ar and gcc-ranlib to support LTO
-    set(CMAKE_AR "gcc-ar")
-    set(CMAKE_RANLIB "gcc-ranlib")
-  endif()
-  if(USING_TI)
-    jerry_add_link_flags(-lto)
-  endif()
+  include(CheckIPOSupported)
+  check_ipo_supported()
+  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
 endif()
 
 # Compiler / Linker flags
-- 
2.45.0.windows.1

