From 4eeef188265a250a9a6dc8656c952bbb94e0a62b Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Mon, 23 Jun 2025 19:53:00 +0800
Subject: [PATCH] conda build

---
 src/opt.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/opt.rs b/src/opt.rs
index 45681f8..5a6bdb1 100644
--- a/src/opt.rs
+++ b/src/opt.rs
@@ -348,7 +348,8 @@ impl CommonData {
         }
     }
 
-    fn from_opts_and_config(value: CommonOpts, config: ConfigBase<()>) -> Self {
+    #[allow(missing_docs)]
+    pub fn from_opts_and_config(value: CommonOpts, config: ConfigBase<()>) -> Self {
         Self::new(
             value.output_dir,
             value.experimental,
-- 
2.49.0.windows.1

