From 4eeef188265a250a9a6dc8656c952bbb94e0a62b Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Mon, 23 Jun 2025 19:53:00 +0800
Subject: [PATCH] conda build

---
 Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index aa7f6da..e5da61c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -182,8 +182,8 @@ retry-policies = "0.5.1"
 strum = "0.27.1"
 unicode-normalization = "0.1.24"
 line-ending = "1.5.1"
-diffy = { git = "https://github.com/prefix-dev/diffy.git", branch = "master" }
-http-range-client = { git = "https://github.com/pka/http-range-client.git", default-features = false, features = [
+diffy = { path = "../diffy" }
+http-range-client = { path = "../http-range-client", default-features = false, features = [
   "reqwest-sync",
   "rustls-tls",
 ] }
-- 
2.49.0.windows.1

