From 4eeef188265a250a9a6dc8656c952bbb94e0a62b Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Mon, 23 Jun 2025 19:53:00 +0800
Subject: [PATCH] conda build

---
 Cargo.toml | 4 ++--
 src/opt.rs | 3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index b8cbd6a..ca54961 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -185,8 +185,8 @@ retry-policies = "0.5.1"
 strum = "0.27.2"
 unicode-normalization = "0.1.24"
 line-ending = "1.5.1"
-diffy = { git = "https://github.com/prefix-dev/diffy.git", branch = "master" }
-http-range-client = { git = "https://github.com/pka/http-range-client.git", default-features = false, features = [
+diffy = { path = "../diffy" }
+http-range-client = { path = "../http-range-client", default-features = false, features = [
   "reqwest-sync",
   "rustls-tls",
 ] }
diff --git a/src/opt.rs b/src/opt.rs
index 212d5c6..7de34af 100644
--- a/src/opt.rs
+++ b/src/opt.rs
@@ -301,7 +301,8 @@ impl CommonData {
         }
     }
 
-    fn from_opts_and_config(value: CommonOpts, config: Config) -> Self {
+    #[allow(missing_docs)]
+    pub fn from_opts_and_config(value: CommonOpts, config: Config) -> Self {
         Self::new(
             value.output_dir,
             value.experimental,
-- 
2.49.0.windows.1

