From be5a6e74f38f1d0d77a4454f70b7dd2964b1c57c Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Fri, 26 Sep 2025 19:10:22 +0800
Subject: [PATCH] pixi

---
 Cargo.toml | 52 ++++++++++++++++++++++++++--------------------------
 1 file changed, 26 insertions(+), 26 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 5ae7318..9b86cd7 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -174,32 +174,32 @@ tracing-subscriber = "=0.3.19"
 typed-path = "0.11.0"
 # Bumping this to a higher version breaks the Windows path handling.
 url = "2.5.4"
-uv-auth = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-build-frontend = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-cache = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-cache-info = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-client = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-configuration = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-dispatch = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-distribution = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-distribution-filename = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-distribution-types = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-git = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-git-types = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-install-wheel = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-installer = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-normalize = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-pep440 = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-pep508 = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-platform-tags = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-pypi-types = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-python = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-redacted = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-requirements = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-requirements-txt = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-resolver = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-types = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
-uv-workspace = { git = "https://github.com/astral-sh/uv", tag = "0.8.5" }
+uv-auth = { path = "../uv/crates/uv-auth" }
+uv-build-frontend = { path = "../uv/crates/uv-build-frontend" }
+uv-cache = { path = "../uv/crates/uv-cache" }
+uv-cache-info = { path = "../uv/crates/uv-cache-info" }
+uv-client = { path = "../uv/crates/uv-client" }
+uv-configuration = { path = "../uv/crates/uv-configuration" }
+uv-dispatch = { path = "../uv/crates/uv-dispatch" }
+uv-distribution = { path = "../uv/crates/uv-distribution" }
+uv-distribution-filename = { path = "../uv/crates/uv-distribution-filename" }
+uv-distribution-types = { path = "../uv/crates/uv-distribution-types" }
+uv-git = { path = "../uv/crates/uv-git" }
+uv-git-types = { path = "../uv/crates/uv-git-types" }
+uv-install-wheel = { path = "../uv/crates/uv-install-wheel" }
+uv-installer = { path = "../uv/crates/uv-installer" }
+uv-normalize = { path = "../uv/crates/uv-normalize" }
+uv-pep440 = { path = "../uv/crates/uv-pep440" }
+uv-pep508 = { path = "../uv/crates/uv-pep508" }
+uv-platform-tags = { path = "../uv/crates/uv-platform-tags" }
+uv-pypi-types = { path = "../uv/crates/uv-pypi-types" }
+uv-python = { path = "../uv/crates/uv-python" }
+uv-redacted = { path = "../uv/crates/uv-redacted" }
+uv-requirements = { path = "../uv/crates/uv-requirements" }
+uv-requirements-txt = { path = "../uv/crates/uv-requirements-txt" }
+uv-resolver = { path = "../uv/crates/uv-resolver" }
+uv-types = { path = "../uv/crates/uv-types" }
+uv-workspace = { path = "../uv/crates/uv-workspace" }
 which = "8.0.0"
 winapi = { version = "0.3.9", default-features = false }
 xxhash-rust = "0.8.15"
-- 
2.51.0.windows.1

