From d14ebaa77193ff3df1cb333857ebfd86979a1604 Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Mon, 4 Nov 2024 20:06:38 +0800
Subject: [PATCH] customize

---
 Cargo.toml | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 3d50a39..316c184 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -3,7 +3,7 @@ members = ["crates/*", "tools/*"]
 resolver = "2"
 
 # See: https://docs.rs/insta/latest/insta/#optional-faster-runs
-[profile.dev.package.insta]
+[profile.release.package.insta]
 opt-level = 3
 
 [workspace.package]
@@ -19,8 +19,11 @@ allow-branch = ["main"]
 consolidate-commits = true
 tag-prefix = ""
 
-[profile.bench]
+[profile.release]
+codegen-units = 1
 lto = true
+opt-level = 3
+strip = true
 
 [workspace.dependencies]
 anyhow = "1.0.82"
-- 
2.47.0.windows.1

