From 5319c98683df18a840bd21fff22d2781b1501627 Mon Sep 17 00:00:00 2001
From: snekdesign <104812819+snekdesign@users.noreply.github.com>
Date: Sat, 22 Jun 2024 19:51:06 +0800
Subject: [PATCH] conda build

---
 Source/Native/NSudoDevilMode/NSudoDevilMode.vcxproj | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Source/Native/NSudoDevilMode/NSudoDevilMode.vcxproj b/Source/Native/NSudoDevilMode/NSudoDevilMode.vcxproj
index 9c3f6e0..1f3018d 100644
--- a/Source/Native/NSudoDevilMode/NSudoDevilMode.vcxproj
+++ b/Source/Native/NSudoDevilMode/NSudoDevilMode.vcxproj
@@ -11,7 +11,6 @@
   </PropertyGroup>
   <Import Project="..\Mile.Cpp\Mile.Project\Mile.Project.props" />
   <ImportGroup Label="PropertySheets">
-    <Import Project="..\libkcrt\libkcrt.props" />
     <Import Project="..\MINT\MINT.props" />
   </ImportGroup>
   <ItemDefinitionGroup>
@@ -20,8 +19,8 @@
       <PreprocessorDefinitions>NSudoDevilMode_EXPORTS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <Link>
-      <IgnoreAllDefaultLibraries>true</IgnoreAllDefaultLibraries>
-      <AdditionalDependencies>ntdll.lib;libkcrt.lib</AdditionalDependencies>
+      <AdditionalLibraryDirectories>$(LIBRARY_PREFIX)\x86_64-w64-mingw32\sysroot\usr\lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalDependencies>kernel32.lib;libntoskrnl.a;ntdll.lib</AdditionalDependencies>
     </Link>
   </ItemDefinitionGroup>
   <ItemGroup>
-- 
2.45.0.windows.1

